runtime/jcl/common/clsldr.cpp

```diff
@@ -107,9 +107,6 @@ Java_java_lang_ClassLoader_defineClassImpl1(JNIEnv *env, jobject receiver, jclas
 	vmFuncs->internalExitVMToJNI(currentThread);
 	if (J9_ARE_ALL_BITS_SET(flags, CLASSOPTION_FLAG_HIDDEN)) {
 		options |= (J9_FINDCLASS_FLAG_HIDDEN | J9_FINDCLASS_FLAG_UNSAFE);
-	} else {
-		/* Validate the name for a normal (non-hidden) class. */
-		validateName = TRUE;
```

Comment:
How do we know removing this isn't going to break something in jdk17 - jdk20? Can we restrict this change to jdk21+?

Reply:
For the failing case, is there another flag besides CLASSOPTION_FLAG_HIDDEN that should be checked to avoid validating the name?

Reply:
For the failure noted above, the `flags` argument is 4 (`MethodHandleNatives.Constants.STRONG_LOADER_LINK`).

Reply:
> How do we know removing this isn't going to break something in jdk17 - jdk20? Can we restrict this change to jdk21+?

As it says in the description, the modified function is only available to privileged code, so it shouldn't prevent jdk17-jdk20 from doing anything that it being done now, but perhaps restricting the change to jdk21+ is a good idea.

Reply:
If the only case that needs to avoid validation is STRONG_LOADER_LINK, we should check for that, and validate otherwise. I don't know all the cases where the API is used and where the input can come from.

Reply:
If the change in openjdk is backported to earlier versions, the code change here can be adopted for those versions.

Reply:
Made version specific in https://github.com/eclipse-openj9/openj9/pull/17126/commits/c7440154944ecaddf61911f21fa39bd550c6718d.

Reply:
> If the only case that needs to avoid validation is STRONG_LOADER_LINK

That isn't the only case, `HIDDEN_CLASS` is also used. The only (current) use of this API in jdk21 is in  `MethodHandles` where it [asserts](https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/openj9/src/java.base/share/classes/java/lang/invoke/MethodHandles.java#L2432) that at least one of those flags is present. However, the recent change means we must not insist on a valid class name (as we used to do) only if `HIDDEN_CLASS` is specified.

Reply:
I meant the only new case. Class name validation is already skipped for hidden.

Reply:
I don't understand your comment. We previously bypassed the name check when `HIDDEN_CLASS` was present; with this we don't do the name check even if that flag is not present (there are only two cases, that bit is set or it's not). I think `STRONG_LOADER_LINK` is completely unrelated to the name of the class.